p4app-exercises/p4runtime.p4app/README.md

```diff
@@ -1,174 +1,183 @@
-# Implementing Basic Forwarding
+# Implementing a Control Plane using P4Runtime
 
 ## Introduction
 
-The objective of this exercise is to write a P4 program that
-implements basic forwarding. To keep things simple, we will just
-implement forwarding for IPv4.
-
-With IPv4 forwarding, the switch must perform the following actions
-for every packet: (i) update the source and destination MAC addresses,
-(ii) decrement the time-to-live (TTL) in the IP header, and (iii)
-forward the packet out the appropriate port.
+In this exercise, we will be using P4Runtime to send flow entries to the 
+switch instead of using the switch's CLI. We will be building on the same P4
+program that you used in the [basic_tunnel](../basic_tunnel) exercise. The
```

Comment:
If I understand correctly, this exercise is purely about P4Runtime, with no modifications at all required to the P4 program. Maybe we should make this clear as this is how the exercise differs from the other 2.
IMO, this is also the reason why you chose to pull the mycontroller.py functionality out of main.py. It makes it "inconsistent" with the other exercises in a way, but it also means that students don't have to run `make run` systematically since they are not changing the P4. Instead, they can make incremental edits to mycontroller.py and just run `make mycontroller`. Is that correct?

Reply:
The primary reason I removed the execution of mycontroller.py out of main.py was laziness -- it makes the README instructions that were already written for exercises/p4runtime/README.md copyable, and require fewer changes to make them true.  They recommended using `make` to start mininet, up to the mininet prompt, then use `h1 ping h2` at the mininet CLI to start a traffic flow that could be observed via counter updates when traffic is flowing as desired.    The way `main.py` is without these changes, there is no mininet prompt, so no way to run `h1 ping h2` at a mininet prompt.

The exercises/p4runtime/ version of this exercise also had the property that you could start mininet once, and quit, edit, and restart the mycontroller.py script as often as you wanted, so it seemed reasonable to preserve that in this version of the exercise.